spec/unit/puppet/type/wait_for_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+
+describe Puppet::Type.type(:wait_for) do
+  let(:wait_for) do
+    Puppet::Type.type(:wait_for).new(:query => 'echo foo bar', :regex => 'foo')
+  end
+
+  test_data = {
+
+    # Illustrating some consequences of acceptable input values due
+    # to the type's data type coercion.
+    #
+    :exit_code   => [[42.5, 42], [[42], 42], [42, 42], ['42', 42]],
+    :seconds     => [[42.5, 42], [[42], 42], [42, 42], ['42', 42]],
+    :max_retries => [[42.5, 42], [42, 42], ['42', 42]],
+    :polling_frequency => [[1.5, 1.5], [1, 1.0]],
+    :regex             => [[/foo/, /foo/], ['foo', /foo/]],
+  }
+
+  test_data.each do |k,v|
+    v.each do |val|
+      i, o = val
+      it "accepts #{k}=>#{i}" do
+        wait_for[k] = i
+        expect(wait_for[k]).to eq o
+      end
+    end
+  end
+
+  it 'does not raise error' do
+    expect { wait_for }.not_to raise_error
+  end
+
+  bad_opts = [
+    {:query  => 'echo foo bar', :regex  => 'foo', :exit_code => 42},
```

Comment:
Note that this illegal set of options, also mentioned in the README I notice, was being allowed through by the type apparently in error, until I corrected the validation logic.